### ISSUE TITLE
How do I OCR a document in Hebrew?

### ISSUE THREAD

**Issue by DoronCohen** [danielquinn/paperless#273]
Hi!

When uploading a document in Hebrew, I get gibberish although I have configured paperless to install Hebrew support. I saw in some discussion that you noted the OCR first tries English and then guesses the language but there is no way it would do that with Hebrew.
Is there a way to configure first OCR language? or maybe try both?

Thanks

**Comment by MasterofJOKers** [comment]
Do you have any interesting messages in the log? E.g. message about language detection.

As far as I can see, paperless first uses the language from `settings.OCR_LANGUAGE` and then tries again with the guessed language, if the guessed language doesn't fit the set language.

If you'd remove hebrew from `src/paperless_tesseract/languages.py` and set `settings.FORGIVING_OCR` to `True`, it should keep the result of its first try, i.e. with the language you set. (It's just a hack, but let's just try to figure out, if that produces a result.)

For a real fix, one might implement a setting to skip language detection and just use the given language.

**Comment by danielquinn** [comment]
The problem is that Paperless isn't smart enough to just *know* what language the document is in, so it tries to guess by OCRing a page out of the document and feeding that text to a guessing library.  That initial OCR is done with whatever you've set `PAPERLESS_OCR_LANGUAGE` to.  The default is `eng`.

So, if you want to parse a Hebrew document, you should set `PAPERLESS_OCR_LANGUAGE` in `/etc/paperless.conf` or in your environment before (re)starting the consumer.  Unfortunately, there's currently no way for Paperless to make this switch inteligently.  I suppose you could run parallel consumers using different environment variables, but if you go that route, you'll want to use a real database and not Sqlite or you'll run the risk of mangling your data.

TL;DR: Set PAPERLESS_OCR_LANGUAGE, restart the consumer, and give that a shot.

**Comment by DoronCohen** [comment]
Thanks for the answer. I will try that in a couple if days.

**Comment by danielquinn** [comment]
I'm going to go ahead and close this and assume everything is alright now.  If I'm wrong about this, feel free to re-open.
